### ISSUE TITLE
File upload filename options trigger an exception

### ISSUE THREAD

**Issue by uandco** [verbb/formie#1183]
### Describe the bug

Using at least one of "Submission ID" or "Submission Date" in the filename options of a file upload field generates an error when loading the form in the frontend:

<img width="584" alt="image" src="https://user-images.githubusercontent.com/26996169/202636274-99512b46-aff7-4659-ab28-2262545a46dd.png">

<img width="654" alt="image" src="https://user-images.githubusercontent.com/26996169/202636323-4ad12335-ef44-4fbf-a3e5-c4e73137e7f4.png">

### Steps to reproduce

1. Add a file upload field
2. Use "Submission ID" in the filename options
3. Load the form in the frontend

### Form settings

- Multi-page form: Yes
- Submission Method: Ajax
- Client-side Validation: Yes
- Custom Form Templates: No

### Craft CMS version

Craft Pro 4.3.3

### Plugin version

dev-craft-4

### Multi-site?

No

**Comment by engram-design** [comment]
Can't seem to replicate this on my end. Are you able to send through the full stack trace error? I'm not sure where `documents` as a field handle is coming from for this error.

**Comment by uandco** [comment]
`documents` is the handle of the file upload field.

Here's the trace:

```
craft\errors\InvalidFieldException: Invalid field handle: documents in /app/vendor/craftcms/cms/src/base/Element.php:4876
Stack trace:
#0 /app/vendor/craftcms/cms/src/base/Element.php(3848): craft\base\Element->normalizeFieldValue('documents')
#1 /app/vendor/verbb/formie/src/fields/formfields/FileUpload.php(538): craft\base\Element->getFieldValue('documents')
#2 /app/vendor/craftcms/cms/src/base/Element.php(4731): verbb\formie\fields\formfields\FileUpload->afterElementSave(Object(verbb\formie\elements\Submission), true)
#3 /app/vendor/verbb/formie/src/elements/NestedFieldRow.php(287): craft\base\Element->afterSave(true)
#4 /app/vendor/craftcms/cms/src/services/Elements.php(2999): verbb\formie\elements\NestedFieldRow->afterSave(true)
#5 /app/vendor/craftcms/cms/src/services/Elements.php(1039): craft\services\Elements->_saveElementInternal(Object(verbb\formie\elements\NestedFieldRow), false, false, true)
#6 /app/vendor/verbb/formie/src/services/NestedFields.php(279): craft\services\Elements->saveElement(Object(verbb\formie\elements\NestedFieldRow), false)
#7 /app/vendor/verbb/formie/src/base/NestedFieldTrait.php(352): verbb\formie\services\NestedFields->saveElements(Object(verbb\formie\fields\formfields\Group), Object(verbb\formie\elements\Submission))
#8 /app/vendor/craftcms/cms/src/base/Element.php(4731): verbb\formie\fields\formfields\Group->afterElementSave(Object(verbb\formie\elements\Submission), true)
#9 /app/vendor/verbb/formie/src/elements/Submission.php(1009): craft\base\Element->afterSave(true)
#10 /app/vendor/craftcms/cms/src/services/Elements.php(2999): verbb\formie\elements\Submission->afterSave(true)
#11 /app/vendor/craftcms/cms/src/services/Elements.php(1039): craft\services\Elements->_saveElementInternal(Object(verbb\formie\elements\Submission), false, false, true)
#12 /app/vendor/verbb/formie/src/controllers/SubmissionsController.php(591): craft\services\Elements->saveElement(Object(verbb\formie\elements\Submission), false)
#13 [internal function]: verbb\formie\controllers\SubmissionsController->actionSubmit()
#14 /app/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#15 /app/vendor/yiisoft/yii2/base/Controller.php(178): yii\base\InlineAction->runWithParams(Array)
#16 /app/vendor/yiisoft/yii2/base/Module.php(552): yii\base\Controller->runAction('submit', Array)
#17 /app/vendor/craftcms/cms/src/web/Application.php(301): yii\base\Module->runAction('formie/submissi...', Array)
#18 /app/vendor/craftcms/cms/src/web/Application.php(625): craft\web\Application->runAction('formie/submissi...', Array)
#19 /app/vendor/craftcms/cms/src/web/Application.php(280): craft\web\Application->_processActionRequest(Object(craft\web\Request))
#20 /app/vendor/yiisoft/yii2/base/Application.php(384): craft\web\Application->handleRequest(Object(craft\web\Request))
#21 /app/web/index.php(26): yii\base\Application->run()
#22 {main}
```

**Comment by uandco** [comment]
OK I managed to replicate it in a simpler form, here's the config:

```
{
    "handle": "test",
    "settings": "{\"displayFormTitle\":false,\"displayCurrentPageTitle\":false,\"displayPageTabs\":false,\"displayPageProgress\":false,\"scrollToTop\":true,\"progressPosition\":\"end\",\"defaultLabelPosition\":\"verbb\\\\formie\\\\positions\\\\AboveInput\",\"defaultInstructionsPosition\":\"verbb\\\\formie\\\\positions\\\\AboveInput\",\"submitMethod\":\"page-reload\",\"submitAction\":\"message\",\"submitActionTab\":null,\"submitActionUrl\":null,\"submitActionFormHide\":false,\"submitActionMessage\":\"[{\\\"type\\\":\\\"paragraph\\\",\\\"attrs\\\":{\\\"textAlign\\\":\\\"start\\\"},\\\"content\\\":[{\\\"type\\\":\\\"text\\\",\\\"text\\\":\\\"Submission saved.\\\"}]}]\",\"submitActionMessageTimeout\":null,\"submitActionMessagePosition\":\"top-form\",\"loadingIndicator\":null,\"loadingIndicatorText\":null,\"validationOnSubmit\":true,\"validationOnFocus\":false,\"errorMessage\":\"[{\\\"type\\\":\\\"paragraph\\\",\\\"attrs\\\":{\\\"textAlign\\\":\\\"start\\\"},\\\"content\\\":[{\\\"type\\\":\\\"text\\\",\\\"text\\\":\\\"Couldn\u2019t save submission due to errors.\\\"}]}]\",\"errorMessagePosition\":\"top-form\",\"requireUser\":false,\"requireUserMessage\":\"[{\\\"type\\\":\\\"paragraph\\\",\\\"attrs\\\":{\\\"textAlign\\\":\\\"start\\\"}}]\",\"scheduleForm\":false,\"scheduleFormStart\":null,\"scheduleFormEnd\":null,\"scheduleFormPendingMessage\":\"[{\\\"type\\\":\\\"paragraph\\\",\\\"attrs\\\":{\\\"textAlign\\\":\\\"start\\\"}}]\",\"scheduleFormExpiredMessage\":\"[{\\\"type\\\":\\\"paragraph\\\",\\\"attrs\\\":{\\\"textAlign\\\":\\\"start\\\"}}]\",\"limitSubmissions\":false,\"limitSubmissionsNumber\":null,\"limitSubmissionsType\":\"total\",\"limitSubmissionsMessage\":\"[{\\\"type\\\":\\\"paragraph\\\",\\\"attrs\\\":{\\\"textAlign\\\":\\\"start\\\"}}]\",\"integrations\":{\"duplicate\":{\"enabled\":\"1\",\"showAllPages\":\"\"},\"honeypot\":{\"enabled\":\"1\",\"showAllPages\":\"\"},\"javascript\":{\"enabled\":\"1\",\"showAllPages\":\"\"}},\"submissionTitleFormat\":\"{timestamp}\",\"collectIp\":false,\"collectUser\":false,\"dataRetention\":null,\"dataRetentionValue\":null,\"fileUploadsAction\":null,\"redirectUrl\":null,\"defaultEmailTemplateId\":null,\"disableCaptchas\":false}",
    "submitActionEntryId": null,
    "submitActionEntrySiteId": null,
    "defaultStatusId": 1,
    "dataRetention": "forever",
    "dataRetentionValue": null,
    "userDeletedAction": "retain",
    "fileUploadsAction": "retain",
    "fieldLayoutId": 42,
    "title": "Test",
    "formTemplate": {
        "name": "UU",
        "handle": "uu",
        "template": "",
        "useCustomTemplates": 0,
        "outputCssLayout": 1,
        "outputCssTheme": 0,
        "outputJsBase": 1,
        "outputJsTheme": 1,
        "outputCssLocation": "page-header",
        "outputJsLocation": "page-footer",
        "sortOrder": null
    },
    "notifications": [],
    "pages": [
        {
            "settings": {
                "submitButtonLabel": "Submit",
                "backButtonLabel": "Back",
                "showBackButton": false,
                "saveButtonLabel": "Save",
                "showSaveButton": false,
                "saveButtonStyle": "link",
                "buttonsPosition": "left",
                "cssClasses": null,
                "containerAttributes": null,
                "inputAttributes": null,
                "enableNextButtonConditions": false,
                "nextButtonConditions": [],
                "enablePageConditions": false,
                "pageConditions": [],
                "enableJsEvents": false,
                "jsGtmEventOptions": []
            },
            "userCondition": null,
            "elementCondition": null,
            "label": "Page 1",
            "rows": [
                {
                    "fields": [
                        {
                            "label": "testUpload",
                            "handle": "testupload",
                            "type": "verbb\\formie\\fields\\formfields\\FileUpload",
                            "settings": {
                                "instructions": null,
                                "required": false,
                                "searchable": true,
                                "sizeLimit": null,
                                "sizeMinLimit": null,
                                "limitFiles": null,
                                "restrictFiles": true,
                                "allowedKinds": [
                                    "image",
                                    "pdf"
                                ],
                                "uploadLocationSource": "volume:465cd49c-4f72-4e1a-a7d7-951b9c6c1347",
                                "uploadLocationSubpath": null,
                                "restrictLocation": true,
                                "filenameFormat": "{submissionId}",
                                "restrictedLocationSource": "volume:465cd49c-4f72-4e1a-a7d7-951b9c6c1347",
                                "restrictedLocationSubpath": "",
                                "allowSubfolders": false,
                                "restrictedDefaultUploadSubpath": null,
                                "defaultUploadLocationSource": null,
                                "defaultUploadLocationSubpath": null,
                                "allowUploads": true,
                                "showUnpermittedVolumes": false,
                                "showUnpermittedFiles": false,
                                "previewMode": "full",
                                "matchField": null,
                                "placeholder": null,
                                "defaultValue": null,
                                "prePopulate": null,
                                "errorMessage": null,
                                "labelPosition": null,
                                "instructionsPosition": null,
                                "cssClasses": null,
                                "containerAttributes": null,
                                "inputAttributes": null,
                                "includeInEmail": true,
                                "enableConditions": false,
                                "conditions": null,
                                "enableContentEncryption": false,
                                "visibility": null,
                                "isNested": false,
                                "limitOptions": null,
                                "displayType": "dropdown",
                                "labelSource": "title",
                                "orderBy": "title ASC",
                                "multiple": false,
                                "allowSelfRelations": false,
                                "localizeRelations": false,
                                "maxRelations": null,
                                "minRelations": null,
                                "selectionLabel": null,
                                "showSiteMenu": false,
                                "source": null,
                                "sources": "*",
                                "targetSiteId": null,
                                "validateRelatedElements": false,
                                "viewMode": null
                            }
                        }
                    ]
                },
                {
                    "fields": [
                        {
                            "label": "test group",
                            "handle": "testGroup",
                            "type": "verbb\\formie\\fields\\formfields\\Group",
                            "settings": {
                                "instructions": null,
                                "required": false,
                                "contentTable": "{{%fmc_1058_testgroup}}",
                                "matchField": null,
                                "placeholder": null,
                                "defaultValue": null,
                                "prePopulate": null,
                                "errorMessage": null,
                                "labelPosition": null,
                                "instructionsPosition": null,
                                "cssClasses": null,
                                "containerAttributes": null,
                                "inputAttributes": null,
                                "includeInEmail": true,
                                "enableConditions": false,
                                "conditions": null,
                                "enableContentEncryption": false,
                                "visibility": null,
                                "isNested": false
                            },
                            "rows": [
                                {
                                    "fields": [
                                        {
                                            "label": "TestUploadInGroup",
                                            "handle": "testuploadingroup",
                                            "type": "verbb\\formie\\fields\\formfields\\FileUpload",
                                            "settings": {
                                                "instructions": null,
                                                "required": false,
                                                "searchable": true,
                                                "sizeLimit": 50,
                                                "sizeMinLimit": 0,
                                                "limitFiles": 10,
                                                "restrictFiles": true,
                                                "allowedKinds": [
                                                    "image",
                                                    "pdf"
                                                ],
                                                "uploadLocationSource": "volume:964952d9-8b03-4fcd-953a-325fbbf1a982",
                                                "uploadLocationSubpath": null,
                                                "restrictLocation": true,
                                                "filenameFormat": "{submissionId}",
                                                "restrictedLocationSource": "volume:964952d9-8b03-4fcd-953a-325fbbf1a982",
                                                "restrictedLocationSubpath": "",
                                                "allowSubfolders": false,
                                                "restrictedDefaultUploadSubpath": null,
                                                "defaultUploadLocationSource": null,
                                                "defaultUploadLocationSubpath": null,
                                                "allowUploads": true,
                                                "showUnpermittedVolumes": false,
                                                "showUnpermittedFiles": false,
                                                "previewMode": "full",
                                                "matchField": null,
                                                "placeholder": null,
                                                "defaultValue": null,
                                                "prePopulate": null,
                                                "errorMessage": null,
                                                "labelPosition": null,
                                                "instructionsPosition": null,
                                                "cssClasses": null,
                                                "containerAttributes": null,
                                                "inputAttributes": null,
                                                "includeInEmail": true,
                                                "enableConditions": false,
                                                "conditions": null,
                                                "enableContentEncryption": false,
                                                "visibility": null,
                                                "isNested": true,
                                                "limitOptions": null,
                                                "displayType": "dropdown",
                                                "labelSource": "title",
                                                "orderBy": "title ASC",
                                                "multiple": false,
                                                "allowSelfRelations": false,
                                                "localizeRelations": false,
                                                "maxRelations": null,
                                                "minRelations": null,
                                                "selectionLabel": null,
                                                "showSiteMenu": false,
                                                "source": null,
                                                "sources": "*",
                                                "targetSiteId": null,
                                                "validateRelatedElements": false,
                                                "viewMode": null
                                            }
                                        }
                                    ]
                                }
                            ]
                        }
                    ]
                }
            ]
        },
        {
            "settings": {
                "submitButtonLabel": "Submit",
                "backButtonLabel": "Back",
                "showBackButton": true,
                "saveButtonLabel": "Save",
                "showSaveButton": false,
                "saveButtonStyle": "link",
                "buttonsPosition": "left",
                "cssClasses": null,
                "containerAttributes": null,
                "inputAttributes": null,
                "enableNextButtonConditions": false,
                "nextButtonConditions": [],
                "enablePageConditions": false,
                "pageConditions": [],
                "enableJsEvents": false,
                "jsGtmEventOptions": []
            },
            "userCondition": null,
            "elementCondition": null,
            "label": "New Page",
            "rows": [
                {
                    "fields": [
                        {
                            "label": "TestUploadPage2",
                            "handle": "testuploadpage2",
                            "type": "verbb\\formie\\fields\\formfields\\FileUpload",
                            "settings": {
                                "instructions": null,
                                "required": true,
                                "searchable": true,
                                "sizeLimit": null,
                                "sizeMinLimit": null,
                                "limitFiles": null,
                                "restrictFiles": true,
                                "allowedKinds": [
                                    "image",
                                    "pdf"
                                ],
                                "uploadLocationSource": "volume:465cd49c-4f72-4e1a-a7d7-951b9c6c1347",
                                "uploadLocationSubpath": null,
                                "restrictLocation": true,
                                "filenameFormat": null,
                                "restrictedLocationSource": "volume:465cd49c-4f72-4e1a-a7d7-951b9c6c1347",
                                "restrictedLocationSubpath": "",
                                "allowSubfolders": false,
                                "restrictedDefaultUploadSubpath": null,
                                "defaultUploadLocationSource": null,
                                "defaultUploadLocationSubpath": null,
                                "allowUploads": true,
                                "showUnpermittedVolumes": false,
                                "showUnpermittedFiles": false,
                                "previewMode": "full",
                                "matchField": null,
                                "placeholder": null,
                                "defaultValue": null,
                                "prePopulate": null,
                                "errorMessage": null,
                                "labelPosition": null,
                                "instructionsPosition": null,
                                "cssClasses": null,
                                "containerAttributes": null,
                                "inputAttributes": null,
                                "includeInEmail": true,
                                "enableConditions": false,
                                "conditions": null,
                                "enableContentEncryption": false,
                                "visibility": null,
                                "isNested": false,
                                "limitOptions": null,
                                "displayType": "dropdown",
                                "labelSource": "title",
                                "orderBy": "title ASC",
                                "multiple": false,
                                "allowSelfRelations": false,
                                "localizeRelations": false,
                                "maxRelations": null,
                                "minRelations": null,
                                "selectionLabel": null,
                                "showSiteMenu": false,
                                "source": null,
                                "sources": "*",
                                "targetSiteId": null,
                                "validateRelatedElements": false,
                                "viewMode": null
                            }
                        }
                    ]
                }
            ]
        }
    ]
}

**Comment by engram-design** [comment]
Right, the difference is this is in a Group field. Fixed for the next release. To get this early, change your `verbb/formie` requirement in `composer.json` to:

```
"require": {
  "verbb/formie": "dev-craft-4 as 2.0.17",
  "...": "..."
}
```

Then run `composer update`.

**Comment by uandco** [comment]
Amazing, I just tested and the issue is gone, thanks for the ultra fast fix!

**Comment by engram-design** [comment]
Fixed in [2.0.18](https://github.com/verbb/formie/releases/tag/2.0.18)
